File: 18_06_2024_Number_Of_Rectangles_In_A_Circle.java
class Solution {
    int rectanglesInCircle(int r) {
          int count = 0;
        int diameterSquared = 4 * r * r;
        for (int w = 1; w <= 2 * r; w++) {
            for (int h = 1; h <= 2 * r; h++) {
                if (w * w + h * h <= diameterSquared) {
                    count++;
                }
            }
        }
        
        return count;
    }
};

File: 18_06_2024_Number_Of_Rectangles_In_A_Circle.py
import math

class Solution:

    def rectanglesInCircle(self, r):
        count = 0
        diameter_squared = (2 * r) ** 2

        for length in range(1, 2 * r + 1):
            max_width = int(math.sqrt(diameter_squared - length ** 2))
            
            count += max_width
        
        return count
